pkg/reporter/reporter.go

```diff
@@ -61,13 +66,69 @@ func Post(report inventory.Report, anchoreDetails config.AnchoreInfo) error {
 	return nil
 }
 
+type anchoreVersion struct {
+	API struct {
+		Version string `json:"version"`
+	} `json:"api"`
+}
+
+// This method retrieves the API version from Anchore
+//
+//nolint:gosec
+func getVersion(anchoreDetails config.AnchoreInfo) (int, error) {
+	log.Debug("Detecting Anchore API version")
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: anchoreDetails.HTTP.Insecure},
+	}
+	client := &http.Client{
+		Transport: tr,
+		Timeout:   time.Duration(anchoreDetails.HTTP.TimeoutSeconds) * time.Second,
+	}
+	resp, err := client.Get(anchoreDetails.URL + "/version")
+	if err != nil {
+		return 0, fmt.Errorf("failed to request API version: %w", err)
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode < 200 || resp.StatusCode > 299 {
+		return 0, fmt.Errorf("failed to retrieve API version: %+v", resp)
+	}
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return 0, fmt.Errorf("failed to read API version: %w", err)
+	}
+
+	parsedResp := anchoreVersion{}
+	err = json.Unmarshal(body, &parsedResp)
+	if err != nil {
+		return 0, fmt.Errorf("failed to parse API version: %w", err)
+	}
+
+	// If parsed response does not include {api: {version: "x"}} then assume V1
+	if parsedResp.API.Version == "" {
+		return 1, nil
+	} else {
+		return strconv.Atoi(parsedResp.API.Version)
+	}
+}
+
 func buildURL(anchoreDetails config.AnchoreInfo) (string, error) {
 	anchoreURL, err := url.Parse(anchoreDetails.URL)
 	if err != nil {
 		return "", err
 	}
 
-	anchoreURL.Path += ReportAPIPath
+	if Version == 0 {
```

Comment:
Why do we need the outer scoped `Version`, and this check?

Reply:
Ah wait I see, this is running it the version check once and holding the result in memory

Reply:
if it's necessary to not run the version check each time, then the global is alright. I think it'd be a good idea to memoize based on the input, specifically `anchoreDetails.URL` in case other deployments are passed that might have different versions we don't want to return the wrong value.

Reply:
That URL is just coming from config; does the service support live reloading config?

Reply:
Not that I can tell, looking at config.go!